### DISCUSSION TITLE
Rebrand PCManFM-Qt

### DISCUSSION THREAD

**Discussion by GreenLunar** [lxqt/lxqt#2206] [Ideas]
Hello,

I think it would be good to rename PCManFM to  LXQt Files ot LXFile or something of sort.
The title "PCManFM-Qt File Manager" is too long and confusing for novice users, with a double mention of file manager (first as FM and second as File Manager).

Please accept this post of mine as a good will gesture.

Thank you.

Of Note:
PCManFM (GTK+ and Qt) is one of the top 3 file managers for Linux, which is one of the few usable file managers intended for novice users and was always independent from extra dependencies, even though it was close to LXDE back then.

**Comment by tsujan** [comment]
LXQt's pcmanfm-qt and LXDE's pcmanfm are totally different projects. The former is being actively developed; the latter doesn't belong to LXQt.

Anyway, I've never been comfortable with the name "pcmanfm-qt", especially because it can be mistaken for "pcmanfm". Maybe, at some point (not soon), we should change it. LXFile sounds good but "LX" may be mistaken for LXDE.

**Reply by GreenLunar** [reply]
> LXFile sounds good but "LX" may be mistaken for LXDE.

How so? It's only a couple of letters.  LX may stand for Swiss International Air Lines ("LX" is an IATA airline designator registered to Swiss International Air Lines).

**Comment by stefonarch** [comment]
I admit too that some applications could have better names and  less capitalized characters. LXimage-qt is among them.
An important thing IMO is that those names can be searched for in search engines -  a search with "lxqt file" will never bring up anything useful. Gnome's names (file, web) are real bad examples here.

 A fork of pcmanfm-qt is named "Filer" which doesn't sound bad.  The names should also be DE--independent. So "Qfiler" could be one possibility but isn't very creative. 
Thinking about thunar and dolphin: are those both derivatives of some fish? ;)

**Reply by tsujan** [reply]
> LXimage-qt is among them.

Yes, those names seem to have been chosen hastily, in early days of LXQt.

> So "Qfiler" could be one possibility but isn't very creative.

"Qfiler" would be good for a purely Qt file manager, while pcmanfm-qt uses the power of GLib with a Qt front-end.

I liked @GreenLunar's suggestion but felt something's missing in it.

**Reply by tsujan** [reply]
> Thinking about thunar and dolphin: are those both derivatives of some fish? ;)

To me, Thunar always existed, but Dolphin was started as a lightweight alternative to Konqueror and I clearly remember that they said it wouldn't replace Konqueror. Now, Dolphin isn't lightweight and Konqueror is a remote memory.

**Reply by GreenLunar** [reply]
> I liked @GreenLunar's suggestion but felt something's missing in it.

It was a mere suggestion/draft idea.

**Reply by GreenLunar** [reply]
> > Thinking about thunar and dolphin: are those both derivatives of some fish? ;)
> 
> To me, Thunar always existed, but Dolphin was started as a lightweight alternative to Konqueror and I clearly remember that they said it wouldn't replace Konqueror. Now, Dolphin isn't lightweight and Konqueror is a remote memory.


Konqueror sounds nice, but mostly looks nice due to the K for KDE, like in many KDE and KDE associated products.  Falkon (K instead of C) is also an example.

Perhaps we should use a name of a, ancient god, or animal or a person which the author likes.

**Reply by GreenLunar** [reply]
> A fork of pcmanfm-qt is named "Filer" which doesn't sound bad. The names should also be DE--independent. So "Qfiler" could be one possibility but isn't very creative. Thinking about thunar and dolphin: are those both derivatives of some fish? ;)

There's no problem with LXFile, LXFiler or LXFiles, IMHO.
Falkon, now a KDE affiliated projects, it was started as QupZilla rebranded to Falkon and to this day, it's still desktop independent.

**Reply by tsujan** [reply]
> There's no problem with LXFile, LXFiler or LXFiles, IMHO.

LXFiles sounds good — it was the "s" that I felt was missing. As for mistaking it for an LXDE app, perhaps I was too cautious.

**Reply by GreenLunar** [reply]
> > There's no problem with LXFile, LXFiler or LXFiles, IMHO.
> 
> ... perhaps I was too cautious.

There is no correct answer (:
Write down a name and make a vote or make a choice ;)

**Comment by stefonarch** [comment]
Konqueror  was quite funny, kind of a swiss knife produced in Italy (common place abuse I know) - it would even open canned food :)

**Comment by GreenLunar** [comment]
Swiss Knife, Swiss International Air Lines ...

New suggested name: **Airline** File Manager (Airline in short)

**Comment by GreenLunar** [comment]
Because files are ordered in tree structure, the followings may be used:

**Forests**
(Daintree) Rainforest

**Trees**
Angel Oak
Cannonball
Cypress
Daintree (Rainforest)
Eucalyptus
Fagus (Sylvatica)
Flamboyant
Hardwood
Jacaranda
Maple
Metasequoia (like metadata and tree)
Oak
Olive
Palm
Palmyra
Ponthus (Beech)
Redwoods
Sagano (Bamboo)
Socotra Dragon
Silk Cotton
Wisteria

**Branches**
Boughs
Branch
Ramus
Twigs

**Reply by GreenLunar** [reply]
I like these: Jacaranda, Palmyra, Rainforest, Ramus, Sagano

**Comment by stefonarch** [comment]
Rainforest you cannot google and has a meaning. The name should not have another meaning. Palmyra is a nice name.
Folllowing thunar and dolphin logic: https://en.wikipedia.org/wiki/List_of_fish_common_names

"Filefish" → FileFisher?

EDIT: already taken https://www.softpedia.com/get/File-managers/File-Fisher.shtml

**Reply by tsujan** [reply]
Moreover, we have no reason to name pcmanfm-qt after a tree. A name should somehow reflect what the app is.

**Reply by GreenLunar** [reply]
@tsujan Some trees have beautiful names, you almost can't go wrong .

@stefonarch ~FileZilla*~ or ~FireFly*~ (these are names of defunct Firefox extension, it might worth a shot)

* Firefly is used by an SBC. It used as a brand of a Firefox extension and a Linux OS.
* FileZilla is taken too

**Reply by stefonarch** [reply]
`<name>FM `  should work with all names.

**Reply by stefonarch** [reply]
There's no _reason_ for trees or fish or animals, but there can be found nice names
PenguinFM, OctopusFM, EagleFM ...

**Comment by GreenLunar** [comment]
Why do you put FM at the end of every suggested brand?

**Reply by stefonarch** [reply]
That's just a proposal, see

> A name should somehow reflect what the app is.

> The name should not have another meaning.

**Comment by redtide** [comment]
FWIW even the project would be nice to rebrand if it will go Wayland, like LQDE.
PCManFM-Qt? PCMFM at this point (Q everywhere is boring :P and it keeps the original name with a full acronym style)

**Reply by tsujan** [reply]
> ...if it will go Wayland

pcmanfm-qt has no problem with Wayland. Apps either work on Wayland or not; their names has nothing to do with that.

None of the names proposed here sounded good to me — except for "LXFiles".

**Reply by redtide** [reply]
IDK, LX sounds like LXDE that AFAIK means Lightweight X11 Desktop Environment, and this is not LXDE nor working only under X11.

**Reply by stefonarch** [reply]
"X" in the names is to avoid, IMHO.
 
Btw  I'm also not really happy with "LXimage-qt" and "lxqt-archiver" but things like QImageviewer and QArchiver are already taken too. Qps and ScreenGrab are good names IMO.

**Comment by GreenLunar** [comment]
Gentlemen,

Just rebrand the whole project to any name that doesn't directly
connected to X11 or Wayland.

Call it Sparrow (Bird) or whatever.

> PCManFM-Qt? PCMFM

No. Call it something else, like Maximian, Machismo, Herculius etc.

Make ut as distinctive as possible.

**Reply by tsujan** [reply]
"X" doesn't mean X11. It's X, like in LinuX and UniX.

I don't think we want to go so far in that direction. Changing all names isn't useful, if not harmful. LXQt is like KDE and GNOME: these names may have been chosen for some reasons at first (e.g., GNOME was Gnu Network Object Model Environment), but now, they are just names.

**Reply by novaTopFlex** [reply]
While X is definitely included as the final letter in UniX inspiring the name "Linux," the X in Xorg may be more closely related to X11. And Wayland may not be well supported on non-GNOME desktop environments.

**Comment by GreenLunar** [comment]
Agreed.

Any suggestions for new names for PCManFM?

**Comment by novaTopFlex** [comment]
I agree with the potential need for the rebranding of "PCManFM-Qt File Manager" as the inclusion of the "PCManFM" name may be confusing for users transitioning to the LXQt desktop environment, as "PCManFM" without the Qt normally refers to the default file manager in LXDE (the Light[weight] X Desktop Environment without the Qt).

**Comment by JamesShriner** [comment]
re-branding MAY cause the same user confusion the OP seeks to avoid.  At this point, many users know an application by it's current name.  Coming from LXDE, I was initially confused between PCManFM, and PCManFM-Qt.  I've since learned the differences, and now the "Qt" suffix is a distinguishing moniker for me and, I assume, others.  With that said, the OP comments aren't without merit.

My original confusion was a one-time thing, and once I learned the nomenclature, was good to go.  I think the OP scenario probably applies to many new(er) users.  It's a marketing issue.  Which is only an issue if you're looking to attract new users, which I assume IS a goal?  New users are how a project grows, and possibly how it attracts new developers?

Several good comments above regarding history of LXDE (Lightweight X11 Desktop Environment) and use of X for X11 and/or X.org.  It demonstrates the roots of how some things came to be.  Similarly LXQt is a fusion of the LXDE and RazorQt desktop projects.  Makes sense to those of us who know the history, or were around to experience it.  Not so much to NEW users.  I'm seeing references above regarding the use of "LX" in the software name, some positive, some critical.  Both viewpoints have merit, IMO.  I would only mention that "too much re-branding" is just as confusing.  Once upon a time, NautilusFM was the file manager for Gnome/GTK desktops.  Now it's just Files.  But there's also Caja (which was forked from Nautilus to avoid the GTK3 toolkit, but has since been ported to GTK3).  So now we've got Nautilus, Files, GnomeFiles, and Caja all used interchangeably (correctly,  or incorrectly, based solely on user-perception) to describe the default Gnome file manager.  I haven't even mentioned Nemo, which is Mint's spin of Nautilus/Files for their Cinnamon (GTK3) desktop.

Personally, I'm indifferent to the name.  I'll use the default manager provided by LXQt, regardless of what it's called.  But if a re-brand is in order (and perhaps it is?!), I'll suggest that a focus on the Qt toolkit may be a viable solution.  Perhaps QtFileManager, or QtFM, or something similar.  Less focus on the "lightweight" L-moniker (which falls in line with LXQt distros re-branding LXQT as a "modern" desktop moreso than lightweight; no focus on the "X" moniker, which is likely to be replaced by Wayland in the forseeable future; and complete focus on the Qt underpinnings of the file manager, as well as the function of the application.  I'm pretty sure THAT won't be changing anytime soon.  JMO.....feel free to disregard.

**Comment by GreenLunar** [comment]
@JamesShriner, I like what you wrote.

I need to read it several times more.

> I'll suggest that a focus on the Qt toolkit may be a viable solution.
> Perhaps QtFileManager, or QtFM, or something similar.

There is already QtFM from @rodlie

page: https://qtfm.eu/
repo: https://github.com/rodlie/qtfm

Perhaps Qt-Filer, Qt-FileClerk, Qt-FastFile, Qt-FastFiler,
Qt-FileOffice or Qt-FileDesk.

Overhaul, and especially due to what you wrote, it is a tough task, and
as much as I desire the name change, I don't even attempt to push it,
because I don't know what is the correct approach to do it.

**Comment by GreenLunar** [comment]
Qt-FileTasker (Qt) (File) (Task) (er)

**Comment by stefonarch** [comment]
Marlin, following the fish schema with dolphin,thunar?

>Marlin is a type of fish. It's a popular game fish known for its speed and agility.

**Reply by redtide** [reply]
if I were to switch to a new nomenclature, I would adopt something context-related, in this case I would focus on "Lightweight" (if we want to keep apart X and Qt terms).
It might sound stupid but as example when dealing with files, why not something like "Origami"? (or "Butterfiles" to play with words :P ) . Anyways, the logic should be applied to all applications, not just one (consistency).

**Comment by GreenLunar** [comment]
On Wed, 06 Sep 2023 00:53:22 -0700
Standreas ***@***.***> wrote:

> Marlin, following the fish schema with dolphin,thunar?
>
> >Marlin is a type of fish. It's a popular game fish known for its
> >speed and agility.
>

Marlin is a beautiful name!

**Comment by tsujan** [comment]
Realistically speaking, changing app names would have costs which aren't acceptable with our manpower. This discussion may continue for finding good names, but rebranding is unlikely to happen.

"PCManFM-Qt"  may not be beautiful, but it tells what it should: a file manager based on `libfm-qt`, whose robust foundation was laid by its original author (I say "robust" because I've worked on its code). We might change it only if we have 3 or 4 times our current manpower for a long period of time.

**Comment by GreenLunar** [comment]
On Wed, 06 Sep 2023 06:10:45 -0700
tsujan ***@***.***> wrote:

> Realistically speaking, changing app names would have costs which
> aren't acceptable with our manpower. This discussion may continue for
> finding good names, but rebranding is unlikely to happen.
>

I partly agree.

This discussion can be useful for finding good names, and a possible
gradual path to rebrand PCManFM-Qt.

One of which would be to ask or make packagers to keep PCManFM-Qt in
package title or description.

Example: Marlin (formerly PCManFM-Qt).

> "PCManFM-Qt"  may not be beautiful, but it tells what it should: a
> file manager based on `libfm-qt`, whose robust foundation was laid by
> its original author.

I don't agree with your statement.

I'm focusing on end-users who want to launch a program and use it.

Most of end-users probably don't know what is libfm-qt or libfm or even
a (program) library.

**Comment by Gnomasz** [comment]
> Less focus on the "lightweight" L-moniker (which falls in line with LXQt distros re-branding LXQT as a "modern" desktop moreso than lightweight;

I do not agree. The "About" page of the project emphasizes lightness: "It will not get in your way. It will not hang or slow down your system. It is focused on being a classic desktop with a modern look and feel." And the DE's design is rather conservative – I think that "modern" fits more to Gnome or KDE.

As a user, I hope that being lightweight remains a focus for LXQt. I remember someone from XFCE posting that with their limited numbers they cannot keep it as priority, that it stands in the way of new features.

As such, I'm more with redtide:

> if I were to switch to a new nomenclature, I would adopt something context-related, in this case I would focus on "Lightweight" (if we want to keep apart X and Qt terms).

Or tangential to lightness, names with "L", for example. "Librarian" for the file manager, perhaps?
That is, if a theme should exist to those names. As stefonarch pointed out, Qps and ScreenGrab are already good, memorable names.

**Reply by tsujan** [reply]
> As a user, I hope that being lightweight remains a focus for LXQt.

Always. In addition, we've always tried to keep the code and GUI as uncluttered as possible — somewhere between the two extremes, namely, GNOME and KDE.

**Comment by redtide** [comment]
thinking about it, as is now it should just be LXFM, but even that sounds wrong to me, since I don't agree with the tsujan' idea of "they are just names", the entire project had to be called LQDE, so a LQFM would keep "qt" and "file manager", no need to keep the name of the maker in an app name, it's enough to know it from the AUTHORS and so in the about dialog.
I think in the end the willing of changing completely the app names, which is clearly against the will of the authors AFAICS, might be only a start for a fork if someone is that crazy to create one.

**Comment by Rabcor** [comment]
Well i have some ideas, but honestly we could sling around ideas all day i doubt any consesnsus will ever be reached.

LXQt Related:
lxqtfm
lxqt-files
lxqt-file-manager
lxqtexplorer

Qt Related:
qtexplorer 
qtfiles
qtandfast

Independent:
alacrity
quetzal (after quetzalcoatl)
featherfm
hastefm
flightfm (fast & light fm?)
quickfm
arcadia

**Reply by novaTopFlex** [reply]
Actually, the name `alacritty` is already the name of a terminal emulator on various Linux systems, so why use `alacrity` for this application?

**Reply by Butterfly** [reply]
> Well i have some ideas, but honestly we could sling around ideas all day i doubt any consesnsus will ever be reached.
> 
> LXQt Related: lxqtfm lxqt-files lxqt-file-manager lxqtexplorer

`lxfm`? I just ran a non-scientific `apt-get show` in a Debian Trixie that includes the experimental repository, and it blew back a raspberry. My rationale is I like not having to reach a pinky finger up there any more than necessary to grab that "q". :laughing:

**Reply by Rabcor** [reply]
well, lxqt and lxde are separate projects which is why i included the qt. 

Also I suppose it could be renamed to Dora. (As in Dora the Explorer, eh?)

**Comment by normal-max** [comment]
I like pcmanfm it is already on my muscle memory. 
I use dmenu.
Program already known by its name among many users and is installed on many distributions as default file manager.
If something works - don't "fix" it.

**Reply by Fabioamd87** [reply]
this is true, but I also find myself squeeze my brain every time i install LXDE from a long time thinking: how was the file manager name?
The more complicated is, the more easy is to forget it's name.

**Comment by Fabioamd87** [comment]
Guys the question is simple:

What's the Windows file manager name? **Explorer**
What's the MacOS file manager name? **Finder**
What was the GNOME file manager name? **Nautilus**
What's the CINAMMON file manager name? **Nemo**

In all case the name imply finding something (in our case files) so the choice of name is quite easy.

I also agree for whom suggest to prepend the name with the LX, in an apple iEverything fashion, but in that case you need to rename ALL the software not just a few.

Another thing, the tecnical name of the file manager is not really important, what the user see is the name in the menu, in that case it should be cristal clear that we are opening a File Manager.
